OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -82,11 +82,12 @@ void RescanForTargets(Actor self)
 
 			// If close enough, we don't care about other actors.
 			var target = self.World.FindActorsInCircle(self.CenterPosition, WormInfo.IgnoreNoiseAttackRange)
-				.FirstOrDefault(x => attackTrait.HasAnyValidWeapons(Target.FromActor(x)));
-			if (target != null)
+				.Select(t => Target.FromActor(t))
+				.FirstOrDefault(t => attackTrait.HasAnyValidWeapons(t));
```

Comment:
I don't see it..in case attackTrait.HasAnyValidWeapons(Target.FromActor(x)) was true for first actor, only one FromActor target was created, now it is for all actors.